src/commands/init.ts

```diff
@@ -0,0 +1,167 @@
+
+import {FlaggablePrompt, GeneratorCommand, exec, makeFlags, readPJSON} from '../generator'
+import {validateBin} from '../util'
+
+async function clone(repo: string, location: string): Promise<void> {
+  try {
+    await exec(`git clone https://github.com/oclif/${repo}.git "${location}" --depth=1`)
+  } catch (error) {
+    const err =
+      error instanceof Error
+        ? new Errors.CLIError(error)
+        : new Errors.CLIError('An error occurred while cloning the template repo')
+    throw err
+  }
+}
+
+const FLAGGABLE_PROMPTS = {
+  bin: {
+    message: 'Command bin name the CLI will export:',
+    validate: (d: string) => validateBin(d) || 'Invalid bin name',
+  },
+  'output-dir': {
+    message: 'Directory to build the CLI in:',
+    validate: (d: string) => existsSync(resolve(d)),
+  },
+  'package-manager': {
+    message: 'Select a package manager:',
+    options: ['npm', 'yarn', 'pnpm'],
+    validate: (d: string) => ['npm', 'pnpm', 'yarn'].includes(d) || 'Invalid package manager',
+  },
+} satisfies Record<string, FlaggablePrompt>
+
+export default class Generate extends GeneratorCommand<typeof Generate> {
+  static description = `This will clone the template repo and update package properties. For CommonJS, the 'oclif/hello-world' template will be used and for ESM, the 'oclif/hello-world-esm' template will be used.`
+
+  static examples = [
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli',
+      description: 'Generate a new CLI with prompts for all properties',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --yes',
+      description: 'Automatically accept default values for all prompts',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --module-type CommonJS --author "John Doe"',
+      description: 'Supply answers for specific prompts',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --module-type CommonJS --author "John Doe" --yes',
+      description: 'Supply answers for specific prompts and accept default values for the rest',
+    },
+  ]
+
+  static flaggablePrompts = FLAGGABLE_PROMPTS
+
+  static flags = {
+    ...makeFlags(FLAGGABLE_PROMPTS),
+    'module-type': Flags.option({
+      options: ['CommonJS', 'ESM'],
+    })({}),
+  }
+
+  static summary = 'Initialize a new oclif CLI'
+
+  async run(): Promise<void> {
+    const outputDir = await this.getFlagOrPrompt({defaultValue: './', name: 'output-dir', type: 'input'})
+    const location = resolve(outputDir)
+
+    this.log(`Initializing oclif in ${chalk.green(location)}`)
+
+    const packageJSON = (await readPJSON(location))!
+    if (!packageJSON) {
+      throw new Errors.CLIError(`Could not find a package.json file in ${location}`)
+    }
+
+    const bin = await this.getFlagOrPrompt({
+      defaultValue: location.split(sep).at(-1)!,
+      name: 'bin',
+      type: 'input',
+    })
+
+    let moduleType = this.flags['module-type']
```

Comment:
This logic allows the flag to be respected as-is, then auto-detection, then ask.